Sources/X509/Verifier/OneOfPolicyBuilder.swift

```diff
@@ -0,0 +1,207 @@
+public struct OneOfPolicyBuilder {}
+
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildLimitedAvailability<Policy: VerifierPolicy>(_ component: Policy) -> Policy {
+        component
+    }
+}
+
+// MARK: empty policy
+extension OneOfPolicyBuilder {
+    @usableFromInline
+    struct Empty: VerifierPolicy {
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] { [] }
+
+        @inlinable
+        init() {}
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            .failsToMeetPolicy(reason: "No policies specified in OneOfPolicies block")
+        }
+    }
+
+    @inlinable
+    public static func buildBlock() -> some VerifierPolicy {
+        Empty()
+    }
+}
+
+// MARK: concatenated policies
+extension OneOfPolicyBuilder {
+    @usableFromInline
+    struct Tuple2<First: VerifierPolicy, Second: VerifierPolicy>: VerifierPolicy {
+        @usableFromInline
+        var first: First
+
+        @usableFromInline
+        var second: Second
+
+        @inlinable
+        init(first: First, second: Second) {
+            self.first = first
+            self.second = second
+        }
+
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] {
+            let firstExtensions = first.verifyingCriticalExtensions
+            let secondExtensions = second.verifyingCriticalExtensions
+            return firstExtensions.filter { secondExtensions.contains($0) }
+        }
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            let firstResult = await self.first.chainMeetsPolicyRequirements(chain: chain)
+            switch firstResult {
+            case .meetsPolicy:
+                return .meetsPolicy
+            case .failsToMeetPolicy(let firstReason):
+                let secondResult = await self.second.chainMeetsPolicyRequirements(chain: chain)
+                switch secondResult {
+                case .meetsPolicy:
+                    return .meetsPolicy
+                case .failsToMeetPolicy(let secondReason):
+                    return .failsToMeetPolicy(reason: "\(firstReason) and \(secondReason)")
+                }
+            }
+        }
+    }
+
+    @inlinable
+    public static func buildPartialBlock<Policy: VerifierPolicy>(first: Policy) -> Policy {
+        first
+    }
+
+    @inlinable
+    public static func buildPartialBlock(
+        accumulated: some VerifierPolicy,
+        next: some VerifierPolicy
+    ) -> some VerifierPolicy {
+        Tuple2(first: accumulated, second: next)
+    }
+}
+
+// MARK: if
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildOptional(_ component: (some VerifierPolicy)?) -> some VerifierPolicy {
+        PolicyBuilder.WrappedOptional(component)
+    }
+}
```

Comment:
I think we need a custom implementation here. With he current implementation this will always meet the policy:
```swift
    Verifier(rootCertificates: .init()) {
        OneOfPolicies {
            if false {
                RFC5280Policy(validationTime: Date())
            }
        }
    }
```
but I think this should fail the policy.

We need a custom WrappedOptional type that failsToMeet the policy if `component` is `nil`.